DataDog/chaos-controller
# Add disk pressure feature

### What does this PR do?

It adds the disk pressure disruption to the controller. This disruption can apply throttling on disk read and write operations.

### Motivation

Add a way to do some chaos on disk dependent applications.

### Testing Guidelines

* Delete your actual minikube cluster and re-create one
  * A new ISO file is needed for this feature to work, cf. additional notes
* Apply the demo deployment
  * For testing purpose, you should change the image to `ubuntu` or something similar so you have a shell and the `dd` command inside the pod
* Exec into the pod
* Go to `/mnt/data` and try to write a file
  * `dd if=/dev/zero of=/mnt/data/foo oflag=direct bs=1K count=1K`
  * it should write a 1MB file and give you an idea of the actual rate
* Apply the `config/samples/disk_pressure.yaml` file
* Re-run the `dd` command
  * It should now take a lot of time

### Additional Notes

A new minikube ISO file has been created for this feature because it needs the `CONFIG_BLK_DEV_THROTTLING=y` flag.
